frontend/src/components/ProductView.tsx

```diff
@@ -0,0 +1,100 @@
+import { useProductViewStyles } from "../styles/ProductViewStyles";
+import { useParams } from "react-router-dom";
+import { Paper, Grid, Button } from "@material-ui/core";
+import { useHistory } from "react-router-dom";
+
+let data = [
+  {
+    name: "product1",
+    category: "category1",
+    price: 15,
+    id: 1,
+    description: "huh",
+    image:
+      "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1001.jpg",
+  },
+  {
+    name: "product2",
+    category: "category1",
+    price: 30,
+    id: 2,
+    description: "huh",
+    image:
+      "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1001.jpg",
+  },
+  {
+    name: "product3",
+    category: "category2",
+    price: 10,
+    id: 3,
+    description: "huh",
+    image:
+      "https://sapui5.hana.ondemand.com/test-resources/sap/ui/documentation/sdk/images/HT-1001.jpg",
+  },
+];
+
+interface Product {
```

Comment:
You should extract the interfaces/models into a separate `models` package with one file per model (e.g. `product.tsx`)